components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/observers/CommonConfigDeployer.java

```diff
@@ -53,7 +57,18 @@ public void createdConfigurationContext(ConfigurationContext configurationContex
         }
 
         try {
-            APIUtil.writeDefinedSequencesToTenantRegistry(tenantId);
+            //Check whether GatewayType is synapse before loading sequences into registry
+            String filePath =
+                    CarbonUtils.getCarbonHome() + File.separator + "repository" + File.separator + "conf" +
+                            File.separator + "api-manager.xml";
+
```

Comment:
You can use the APIManagerConfigurationService OSGI service available with out reading api-manager.xml again here.

APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
getAPIManagerConfiguration();